File: sdk/kotlin/src/main/kotlin/vn/hitechcloud/sdk/Pmg.kt
package vn.hitechcloud.sdk

/** HiTechCloudPMG: mail filtering configuration and management */
class PmgResource(private val httpClient: HttpClient) {
    /** GET /api/service/{id}/htcpmg/config - Get mail filtering configuration */
    fun getConfig(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/service/{id}/htcpmg/config", params)
    }

    /** POST /api/service/{id}/htcpmg/domains - Add a domain */
    fun addDomain(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.post("/api/service/{id}/htcpmg/domains", params)
    }

    /** POST /api/service/{id}/htcpmg/transport - Set target mail server */
    fun setTransport(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.post("/api/service/{id}/htcpmg/transport", params)
    }

    /** GET /api/service/{id}/htcpmg/stats - Get mail statistics */
    fun getStats(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/service/{id}/htcpmg/stats", params)
    }

}

File: sdk/swift/Sources/HiTechCloudSDK/Pmg.swift
import Foundation

/// HiTechCloudPMG: mail filtering configuration and management
public class PmgResource {
    private let httpClient: HTTPClient
    public init(httpClient: HTTPClient) { self.httpClient = httpClient }

    /// GET /api/service/{id}/htcpmg/config - Get mail filtering configuration
    public func getConfig(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/service/{id}/htcpmg/config", params: params)
    }

    /// POST /api/service/{id}/htcpmg/domains - Add a domain
    public func addDomain(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.post("/api/service/{id}/htcpmg/domains", params: params)
    }

    /// POST /api/service/{id}/htcpmg/transport - Set target mail server
    public func setTransport(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.post("/api/service/{id}/htcpmg/transport", params: params)
    }

    /// GET /api/service/{id}/htcpmg/stats - Get mail statistics
    public func getStats(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/service/{id}/htcpmg/stats", params: params)
    }

}
